examples/visualization/gradcam_demo.py

```diff
@@ -0,0 +1,83 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+
+from keras_cv import visualization
+from keras_cv.visualization.utils import normalize, collect_endpoints
+
+SIZES = (299, 299)
+LABELS = 5
+
+# Load images from the web
+def load_image(path):
+    image = tf.keras.preprocessing.image.load_img(path, target_size=SIZES)
+    return tf.keras.preprocessing.image.img_to_array(image)
+
+paths = [
+    tf.keras.utils.get_file("african_elephant.jpg", "https://i.imgur.com/Bvro0YD.png"),
+    tf.keras.utils.get_file("wolf.jpg", "https://i.imgur.com/rRcsL2D.jpeg"),
+    tf.keras.utils.get_file("cat_and_dog.jpg", "https://i.imgur.com/6wxtMEp.jpeg"),
+]
+
+images = np.stack([load_image(path) for path in paths]).astype("uint8")
+x = tf.convert_to_tensor(images.astype("float32") / 127.5 - 1)
+
+# Load a model with pre-trained weights from ImageNet
+input_tensor = tf.keras.Input([None, None, 3], name="images")
+rn50 = tf.keras.applications.ResNet50V2(
+    input_tensor=input_tensor, weights="imagenet", classifier_activation=None
+)
+
+# Grad-CAM works by computing the gradient of an score (logit) unit with
+# respect to an intermediate activation positional signal A^k_{ij} in the
+# network.
+#
+# We define a new model that outputs:
+#
+#   (a) The last positional signal of the network (the input tensor of the
+#       Global Average Pooling layer);
+#   (b) The output scores (logits) of the network.
+#
+endpoints = collect_endpoints(
+    rn50,
+    endpoints=[
+        {"name": "avg_pool", "link": "input"},
+        {"name": "predictions"},
+    ],
+)
+
+rn50_w_acts = tf.keras.Model(
+    inputs=rn50.inputs,
+    outputs=endpoints,
+)
+
+# For each image, get the index of the 5 labels with
+# highest associated classification energy:
+logits = rn50.predict(x, verbose=0)
+labels = tf.argsort(logits, axis=-1, direction="DESCENDING")[..., :LABELS]
+probs = tf.nn.softmax(logits)
+decoded = tf.keras.applications.imagenet_utils.decode_predictions(probs.numpy())
+
+logits, maps = visualization.gradcam(rn50_w_acts, x, labels)
+
+# As we are only interested in pixels that positively contribute to the
+# classification of a label, we crunch pixels that negatively contribute to
+# its classification and unrelated pixels (zero contribution) together:
+maps = tf.nn.relu(maps)
+maps = normalize(maps)
```

Comment:
Ah - I see how you use this.  Is there any way to encapsulate this information into the gradcam implementation?

Reply:
There is, but I wouldn't recommend adding it directly to the gradcam traced function.
Some other methods will use gradcam or saliency-grad-backprop internally (for example, Smooth-Grad or cameras).
The former consists of replicating the input image N times and adding noise to the data, computing the saliency map with grad-backprop and averaging the results.

Normalizing this could make it difficult for us to reuse these functions on the long run.

#### Some options:

Opt 2: we could have a wrapper:
```
def explain(model, endpoints, explaining_units, images, method='gradcam', normalize=True):
  method = deserialize(method)
  maps = method(..., images)
  if normalize:
    maps = normalize(maps)
  return maps
```

This simplifies everything into a single call, but introduces a performance issue in case the explanation method must be applied over multiple batches (endpoints are extracted and a new model is synthesized each step).

Opt 3: keras-vis opted for a object-oriented approach:
```
class Vis:
  def __init__(self, normalize=True):
    self.normalize = normalize

  def explain(images):
    images = self.perform(images)
    if self.normalize: images = self.normalize(images)
    return images

class GradCam(Vis):
  def perform(images):
    ...
```
Note: that's not the actual props/methods names, I'm just typing on top of my memory to explain it.

This opt takes care of the wrapping and leaves the actual implementation for the sub-classes. The prepared model (with extracted endpoints) can be cached as a property of the Vis' object and calling explain onto multiple batches should be fine. The issue with this approach is that we get a little further from functional programming and it might be a little confusing for some.